### PR TITLE
Fix for issue 538 to paramaterize file_cache_path references

### PR DESCRIPTION
**Pull request by teknomar7**
### Description

This change creates a node attribute default['java']['download_path'] which defaults to Chef::Config[:file_cache_path].  All references to Chef::Config[:file_cache_path] have been replaced by node['java']['download_path'] which allows wrappers to set the value of the download and extract path of the tarball to a location other than Chef::Config[:file_cache_path].  These references were not replaced in spec files.  

### Issues Resolved

This is a resolution to issue 538. The path to download and extract the tarball was hardcoded to  Chef::Config[:file_cache_path] which caused failures at my organization due to a 500 mb partition sizing for /var/chef across the organization.  

### Check List
Note:  All tests did not pass on the master branch prior to modifying the code, so similar tests failed when running the same tests after code modifications.  

- [ ?] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ x] New functionality includes testing.
- [ x] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by teknomar7** [comment]
Lint test fails due to "accept license" prompt for chef exec that times out.

**Comment by JohnRoesler** [comment]
@teknomar7 I've opened PRs to resolve this https://github.com/sous-chefs/java/pull/555 & https://github.com/sous-chefs/terraform-github-org/pull/180

**Comment by teknomar7** [comment]
@JohnRoesler Will I need to do anything else to retrigger the lint test (or anything further in general)?  I'm a little new to this whole process.  

Thanks!

**Comment by JohnRoesler** [comment]
Once those PRs get merged - this PR will ask us to merge those change in and that should do it 👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,12 @@ This file is used to list changes made in each version of the Java cookbook.
 
 - Upgrade circleci orb to version 2 and add yamllint and markdown lint
 
+## 4.2.0 - 2019-06-20
+
+- Fix for issue 538
+- Added "download_path" node attribute defaulting to file_cache_path
+- Replaced all hardcoded instances of file_cache_path with the node attribute
```
Comment (JohnRoesler) [inline CHANGELOG.md]: markdown linter is complaining of extra trailing spaces here

File: README.md [hunk 2]
```diff
@@ -90,6 +90,7 @@ Chef 13.4+
 
 See `attributes/default.rb` for default values.
 
+- `node['java']['download_path']` - Location to download and extract the tarball
```
Comment (JohnRoesler) [inline README.md]: and trailing spaces here

File: CHANGELOG.md (hunk 1 above)
Comment (JohnRoesler) [reply CHANGELOG.md]: fyi, you can download and run locally https://github.com/markdownlint/markdownlint `mdl .`

**Comment by teknomar7** [comment]
Is there any target date for when this might be included in a release to the master branch?  Just want to keep my org apprised since I wrote a temporary work around in lieu of using this cookbook.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,12 @@ This file is used to list changes made in each version of the Java cookbook.
 
 - Upgrade circleci orb to version 2 and add yamllint and markdown lint
 
+## 4.2.0 - 2019-06-20
```
Comment (JohnRoesler) [inline CHANGELOG.md]: Could you update the date here and update the version in the `metadata.rb` to match 4.2?

**Comment by JohnRoesler** [comment]
@majormoses fancy doing a release for 4.2 once this gets updated and merged?

File: CHANGELOG.md (hunk 3 above)
Comment (teknomar7) [reply CHANGELOG.md]: Version in metadata.rb is already 4.2.0 in my repo... should be reflected here.  Are you asking me to just update the date to today's date then with no other changes?

Comment (JohnRoesler) [reply CHANGELOG.md]: Ah shoot - was looking at a specific commit diff that didn't show the metadata bump 👍 yes, just update the date and then I think it looks get - I'll bug @majormoses to get it released

Comment (teknomar7) [reply CHANGELOG.md]: Done... looks like it needs an approving review again though.

**Comment by majormoses** [comment]
released: https://supermarket.chef.io/cookbooks/java/versions/4.2.0

**Pull request merged**